.circleci/config.yml

```diff
@@ -1,39 +1,68 @@
 version: 2.1
 
-plain-go114: &plain-go114
+plain-go: &plain-go
   working_directory: /home/circleci/dd-trace-go.v1
   docker:
-    - image: circleci/golang:1.14
+    - image: circleci/golang:latest
       environment:
         GOPATH: "/home/circleci/go"
 
 jobs:
-  go1_12-build:
-    # Validate that the core builds with go1.12
+  test-core:
+    parallelism: 4
     parameters:
+      gover:
+        type: string
       build_tags:
-        description: "go build tags used to compile"
+        description: "go build tags to use to compile the tests"
         default: ""
         type: string
+    environment: # environment variables for the build itself
+      TEST_RESULTS: /tmp/test-results # path to where test results will be saved
+    resource_class: xlarge
     docker:
-      - image: circleci/golang:1.12
+      - image: circleci/golang:<< parameters.gover >>
         environment:
           GOPATH: "/home/circleci/go"
     working_directory: /home/circleci/dd-trace-go.v1
     steps:
-    - checkout
+      - checkout
+      - run:
+          name: Setup
+          command: |
+            mkdir -p $TEST_RESULTS
+            cp go.sum go.sum.orig
+      - restore_cache: # restores saved cache if no changes are detected since last run
+          keys:
+            - go-mod-v5-core-<< parameters.gover >>-{{ checksum "go.sum.orig" }}
+      - run:
+          name: Tidy Modules
+          command: GO111MODULE=on go mod tidy || exit 0 # this step can fail and we don't care.
```

Comment:
Q: Why are we okay with tidy failing?

Reply:
NIT: Might be nice to explain this in a comment in the `config.yml` - otherwise it's easy for this kind of background info to get lost.

Reply:
Perhaps we can be a bit more descriptive in the comment as to why it's fine if this fails.